src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -924,26 +919,27 @@ internal static DeepCopier GetCopier(Type t)
         /// </summary>
         /// <param name="original">The input data to be deep copied.</param>
         /// <returns>Deep copied clone of the original input object.</returns>
-        public static object DeepCopy(object original)
+        public object DeepCopy(object original)
         {
+            var context = this.serializationContext.Value;
+            context.Reset();
+            
             Stopwatch timer = null;
             if (StatisticsCollector.CollectSerializationStats)
             {
                 timer = new Stopwatch();
                 timer.Start();
-                Copies.Increment();
+                context.SerializationManager.Copies.Increment();
```

Comment:
accessing the SerializationManager instance from the `context` is a little odd from here. Could it be possible that they differ? If it's not possible, then maybe just assert it in the beginning and then use `this` instead?